vcd/resource_vcd_vapp_vm_extraconfig_test.go

```diff
@@ -0,0 +1,596 @@
+//go:build vapp || vm || standaloneVm || ALL || functional
+
+package vcd
+
+import (
+	"fmt"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/terraform"
+	"github.com/vmware/go-vcloud-director/v2/types/v56"
+	"testing"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/resource"
+)
+
+func TestAccVcdVAppVm_extraconfig(t *testing.T) {
+	preTestChecks(t)
+
+	initialValue := "value"
+	updatedValue := "another value"
+	var params = StringMap{
+		"TestName":          t.Name(),
+		"Org":               testConfig.VCD.Org,
+		"Vdc":               testConfig.Nsxt.Vdc,
+		"Catalog":           testConfig.VCD.Catalog.NsxtBackedCatalogName,
+		"CatalogItem":       testConfig.VCD.Catalog.CatalogItemWithMultiVms,
+		"Media":             testConfig.Media.NsxtBackedMediaName,
+		"NsxtEdgeGateway":   testConfig.Nsxt.EdgeGateway,
+		"ExtraConfigValue1": initialValue + "1",
+		"ExtraConfigValue2": initialValue + "2",
+		"FuncName":          t.Name() + "-step1",
+		"Tags":              "vapp vm",
+		"SkipMessage":       " ",
+	}
+	testParamsNotEmpty(t, params)
+
+	configTextStep1 := templateFill(testAccVcdVAppVm_extraconfig, params)
+
+	params["SkipMessage"] = "# skip-binary-test: only for updates"
+	params["ExtraConfigValue1"] = updatedValue + "1"
+	params["ExtraConfigValue2"] = updatedValue + "2"
+	params["FuncName"] = t.Name() + "-step2"
+	configTextStep2 := templateFill(testAccVcdVAppVm_extraconfig, params)
+
+	params["ExtraConfigValue1"] = ""
+	params["ExtraConfigValue2"] = ""
+	params["FuncName"] = t.Name() + "-step3"
+	configTextStep3 := templateFill(testAccVcdVAppVm_extraconfig, params)
+
+	if vcdShortTest {
+		t.Skip(acceptanceTestsSkipped)
+		return
+	}
+	debugPrintf("#[DEBUG] CONFIGURATION: %s\n", configTextStep1)
```

Comment:
Not sure whether we want to log the other steps as well?

Reply:
Right. Added